src/main/java/com/zombiecastlerush/building/Shop.java

```diff
@@ -0,0 +1,48 @@
+package com.zombiecastlerush.building;
+
+import com.zombiecastlerush.entity.Player;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Shop extends Room {
+
+    public Shop(String name, String description) {
+        super(name, description);
+    }
+
+    public void buy(Player player, Item item) {
```

Comment:
Maybe you should make `buy` into `sellItemToPlayer` and `sell` into `buyItemFromPlayer`. Also, perhaps if you can, eliminate the need for passing reference of player, can you possibly get the player reference from inheritance? If not, it's ok.

Reply:
changed names accordingly